MovieQuiz/Presentation/MovieQuizPresenter.swift

```diff
@@ -0,0 +1,131 @@
+    private let questionsAmount: Int = 10
+    private var currentQuestion: QuizQuestion?
+    private var currentQuestionIndex: Int = 0 
+    var correctAnswers: Int = 0
+    
+    init(viewController: PresenterUseViewController) {
+        self.viewController = viewController
+        statistics = StatisticsServiceImplementation()
+        
+        questionFactory = QuestionFactory(delegate: self, moviesLoader: MoviesLoader())
+        viewController.showLoadingIndicator(parameter: false)
+        questionFactory?.loadData()
+    }
+    
+    private func isLastQuestion() -> Bool {
+        currentQuestionIndex == questionsAmount - 1
+    }
+    func resetQuestionIndex() {
+        currentQuestionIndex = 0
+    }
+    private func switchToNextQuestion() {
+        currentQuestionIndex += 1
+    }
+    // func convert - SHOULD BE PRIVATE (deleted `private` due to `MovieQuizPresenterTests`)
+    // search by Find navigator of `convert` within the project showed, that I can delete this `private`
+    func convert(model: QuizQuestion) -> QuizStepViewModel { // Конвертация вопроса в ViewModel
+        let questionStep = QuizStepViewModel(
+              image: UIImage(data: model.image) ?? UIImage(),
+              question: model.text,
+              questionNumber: "\(currentQuestionIndex+1)/\(questionsAmount)")
+        return questionStep
+    }
+    private func didAnswer(isYes: Bool) {
+        let userAnswer: Bool = isYes
+        guard let currentQuestion = currentQuestion else {
+            return }
+        showAnswerResult(isCorrect: userAnswer == currentQuestion.correctAnswer)
+    }
+    func noButtonClicked() {
+        didAnswer(isYes: false)
+    }
+    func yesButtonClicked() {
+        didAnswer(isYes: true)
+    }
+    private func showAnswerResult(isCorrect: Bool) {
+        if isCorrect {
+            correctAnswers += 1
+            viewController?.frameHighlight(isCorrect)
+        } else {
+            viewController?.frameHighlight(isCorrect)
+        }
+        // через 1 секунду c помощью диспетчера задач
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) { [weak self] in
+            guard let self = self else { return }
+            self.showNextQuestionOrResults()
+        }
+    }
+     func showNextQuestionOrResults() {
+        if self.isLastQuestion() {
+// Предварительная подготовка параметров модели "QuizResultsViewModel"
+            guard let statistics = statistics else {
+                print("statistics object error")
+                return}
+            
+            let currentGame = GameRecord(correct: correctAnswers, total: self.questionsAmount, date: Date())
+            statistics.setBestGame(currentGame)
+          
+            let record: String = """
+                                 \(statistics.bestGame.correct)/\(statistics.bestGame.total) \
+                                 (\(statistics.bestGame.date.dateTimeString))
+                                 """
```

Comment:
строчка не поместилась, кроме разбития ее на несколько строк еще можно попробовать делать локальные переменные 
```suggestion
            let bestGame = statistics.bestGame
            let record: String = "\(bestGame.correct)/\(bestGame.total) (\(bestGamedate.dateTimeString))"
```

Reply:
Спасибо за полезный совет!